test/jasmine/assets/custom_assertions.js

```diff
@@ -60,88 +60,99 @@ exports.assertHoverLabelStyle = function(g, expectation, msg, textSelector) {
 };
 
 function assertLabelContent(label, expectation, msg) {
-    var lines = label.selectAll('tspan');
-    var lineCnt = lines.size();
-    var expectMultiLine = Array.isArray(expectation);
+    if(!expectation) expectation = '';
 
-    function extract(sel) {
-        return sel.node() ? sel.html() : null;
-    }
+    var lines = label.selectAll('tspan.line');
+    var content = [];
 
-    if(lineCnt > 0) {
-        if(expectMultiLine) {
-            expect(lines.size()).toBe(expectation.length, msg + ': # of lines');
-            lines.each(function(_, i) {
-                var l = d3.select(this);
-                expect(extract(l)).toBe(expectation[i], msg + ': tspan line ' + i);
-            });
-        } else {
-            fail('Expected a single-line label, found multiple lines');
+    function fill(sel) {
+        if(sel.node()) {
+            var html = sel.html();
+            if(html) content.push(html);
         }
+    }
+
+    if(lines.size()) {
+        lines.each(function() { fill(d3.select(this)); });
     } else {
-        if(!expectMultiLine) {
-            expect(extract(label)).toBe(expectation, msg + ': text content');
-        } else {
-            fail('Expected a multi-line label, found single');
-        }
+        fill(label);
     }
+
+    expect(content.join('\n')).toBe(expectation, msg + ': text content');
 }
 
 function count(selector) {
     return d3.selectAll(selector).size();
 }
 
+/**
+ * @param {object} expectation
+ *  - nums {string || array of strings}
+ *  - name {string || array of strings}
+ *  - axis {string}
+ * @param {string} msg
```

Comment:
Thanks - I find that a lot easier, especially as you point out since this lives in a different file from where it gets called.